### PR TITLE
Use indexers to access registers

### PR DESCRIPTION
**Pull request by kevinferrare**

### CODE REVIEW

File: src/Spice86.Core/Emulator/CPU/ModRM.cs [hunk 1]
```diff
@@ -18,48 +19,48 @@ public class ModRM {
         _state = state;
     }
 
-    public uint GetAddress(int defaultSegmentRegisterIndex, ushort offset) {
-        int? segmentIndex = _state.SegmentOverrideIndex;
+    public uint GetAddress(uint defaultSegmentRegisterIndex, ushort offset) {
+        uint? segmentIndex = _state.SegmentOverrideIndex;
         segmentIndex ??= defaultSegmentRegisterIndex;
```
Comment (JorisVanEijden) [inline src/Spice86.Core/Emulator/CPU/ModRM.cs:24]:
```suggestion
        uint? segmentIndex = _state.SegmentOverrideIndex ?? defaultSegmentRegisterIndex;
```

File: src/Spice86.Core/Emulator/CPU/ModRM.cs [hunk 2]
```diff
@@ -18,48 +19,48 @@ public class ModRM {
...
 
-        ushort segment = _state.SegmentRegisters.GetRegister16((int)segmentIndex);
+        ushort segment = _state.SegmentRegisters.UInt16[(int)segmentIndex];
```
Comment (JorisVanEijden) [inline src/Spice86.Core/Emulator/CPU/ModRM.cs:26]: Why is a `uint?` cast to an `int`?
I'd expect the segmentIndex to either already be an `int`, or the `segmentIndex.Value` being used.

File: src/Spice86.Core/Emulator/CPU/Registers/UInt32RegistersIndexer.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class UInt32RegistersIndexer : RegistersIndexer<uint> {
+    private readonly IUIntReaderWriter _uIntArrayReaderWriter;
+
+    public UInt32RegistersIndexer(IUIntReaderWriter uIntArrayReaderWriter) {
+        _uIntArrayReaderWriter = uIntArrayReaderWriter;
+    }
+
+    public override uint this[uint index] {
+        get => _uIntArrayReaderWriter[(uint)index];
+        set => _uIntArrayReaderWriter[(uint)index] = value;
```
Comment (JorisVanEijden) [inline src/Spice86.Core/Emulator/CPU/Registers/UInt32RegistersIndexer.cs:17]: Isn't index already an uint?

Comment (kevinferrare) [reply src/Spice86.Core/Emulator/CPU/Registers/UInt32RegistersIndexer.cs:17]: indeed, there are lot of places like that in the code, Indexer actually accepts both uint and int

File: src/Spice86.Core/Emulator/CPU/ModRM.cs (hunk 1 above)
Comment (kevinferrare) [reply src/Spice86.Core/Emulator/CPU/ModRM.cs:24]: Done

**Review by JorisVanEijden — APPROVED** [review]

**Review by maximilien-noal — APPROVED** [review]

**Pull request merged**